halo2_proofs/src/poly/evaluator.rs

```diff
@@ -225,6 +241,19 @@ impl<E, F: Field, B: Basis> Evaluator<E, F, B> {
                     }
                     lhs
                 }
+                Ast::DistributePowers(terms, base) => {
+                    let mut acc = vec![F::zero(); ctx.local_chunk_size()];
+
+                    for term in terms.iter() {
+                        let term = recurse(term, ctx);
+                        for (acc, term) in acc.iter_mut().zip(term) {
+                            *acc *= base;
+                            *acc += term;
+                        }
```

Comment:
I confirmed that this is equivalent to the previous nested stack of `Ast::Scale` and `Ast::Add`.